lib/api/core/models/repositories/tokenRepository.js

```diff
@@ -96,12 +96,12 @@ module.exports = function (kuzzle) {
     try {
       expiresIn = parseTimespan(options.expiresIn);
       encodedToken = jwt.sign({_id: user._id}, kuzzle.config.jsonWebToken.secret, options);
-      this.hydrate(token, {
+      q(_.assignIn(token, {
```

Comment:
This too can be simplified by simply modifying `token` with `_.assignIn()` and then start the promise chain with `persistToCache`:

```js
_.assignIn(token, {
        _id: encodedToken,
        user: user._id,
        ttl: expiresIn,
        expiresAt: Date.now() + expiresIn
});

return this.persistToCache(token)
  .then(() => {
    kuzzle.tokenManager.add(token, context);
    deferred.resolve(token);
  })
  .catch(err => { ...
```